### PR TITLE
Add autocomplete onOpen and onClose hooks

### PR DESCRIPTION
**Pull request by alexisgrow**
TEST=manual

Test with the theme, specifying both onOpen and onClose with simple `console.log` statements. See the "open" print statement print once when the autocomplete opens, and "close" print once when autocomplete closes.

### CODE REVIEW

File: src/ui/components/search/searchcomponent.js [hunk 1]
```diff
@@ -273,6 +273,12 @@ export default class SearchComponent extends Component {
      * @type {string}
      */
     this.inputIdName = `yxt-SearchBar-input--${this.name}`;
+
+    /**
+     * Options to pass to the autocomplete component
+     * @type {Object}
+     */
+    this._autocompleteConfig = config.autocomplete || {};
```
Comment (oshi97) [inline src/ui/components/search/searchcomponent.js]: what kind of config is being passed? I think if possible we want to avoid passing config directly

Comment (oshi97) [reply src/ui/components/search/searchcomponent.js]: took a look at the theme prs and I'm back! would it make sense to just expose the config needed for the overlay? we would probably want to add it to the readme as well

Comment (alexisgrow) [reply src/ui/components/search/searchcomponent.js]: Ah thanks!! I just updated to pass only the specific new config: onOpen and onClose hooks!

**Review by oshi97 — APPROVED** [review]
lgtm!

**Pull request merged**